src/widgets/gl_widget.cpp

```diff
@@ -10,15 +10,14 @@
 
 namespace QMapLibre {
 
-GLWidget::GLWidget(const Settings &settings) {
-    d_ptr = new GLWidgetPrivate(this, settings);
-}
+GLWidget::GLWidget(const Settings &settings)
+    : d_ptr(std::make_unique<GLWidgetPrivate>(this, settings)) {}
 
 GLWidget::~GLWidget() {
     // Make sure we have a valid context so we
     // can delete the Map.
     makeCurrent();
-    delete d_ptr;
+    d_ptr.reset();
```

Comment:
No need to do this, `d_ptr`'s destructor will free the memory.

Reply:
The idea here is to ensure the context is valid. I decided to make it explicit, otherwise the destructor looks a bit weird.